.github/workflows/rust.yml

```diff
@@ -15,25 +21,40 @@ jobs:
         uses: actions/checkout@v2
 
       - name: Set up cargo/rust
-        run: |
-          rustup component add rustfmt
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          components: rustfmt, clippy
+          override: true
 
       - name: Setup OCaml (because of ocaml-gen)
-        run: sudo apt update && sudo apt install ocaml
+        run: |
+          sudo apt update
+          sudo apt install -y ocaml
 
-      - name: build
+      # `cargo check` compiles the packages without performing the final step of code generation,
+      # which is faster than running `cargo build`.
+      # Use release build to speed up tests
+      - name: build & test
         run: |
-          cargo build
+          cargo check --release
+          # This is covered by the next step,
+          # commented out to make the job faster
+          #
+          # cargo test --release
 
-      - name: build ocaml types
+      - name: build & test ocaml types
         run: |
-          cargo build --features ocaml_types
+          # Cannot use `--all-features` here
+          # Becauses `3`, `3w` and `5w` features under `oracle` crate are orthogonal,
+          # Should this be changed?
+          cargo check --release --features ocaml_types
+          cargo test --release --features ocaml_types
```

Comment:
Move this to its own step

Reply:
Fixed https://github.com/o1-labs/proof-systems/pull/332/files#diff-73e17259d77e5fbef83b2bdbbe4dc40a912f807472287f7f45b77e0cbf78792dR40